src/solidity/astHelper.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+var AstWalker = require('../util/astWalker')
+
+/**
+  * return all contract definitions of the given @astList
+  *
+  * @param {Object} astList  - AST nodes of all the sources
+  * @return {Object} - return the AST node of found contracts
+  */
+function getContractsDefinition (astList) {
+  var contracts = {}
+  var walker = new AstWalker()
+  walker.walkAstList(astList, { 'ContractDefinition': function (node) {
+    contracts[node.id] = node
+    return false
+  }})
+  return contracts
+}
+
+/**
+  * returns the linearized base contracts of the contract @arg id
+  *
+  * @param {Int} id  - contract id for which based contracts should be resolved
+  * @param {Map} contracts  - all contracts defined in the current context
+  * @return {Object} - return the AST node of found contract
+  */
+function getBaseContracts (id, contracts) {
+  var baseContracts = contracts[id].attributes.linearizedBaseContracts.map(function (id) { return contracts[id] })
+  baseContracts.reverse()
```

Comment:
Actually I think it is not good practice to reverse at this point, because in all other places in the codebase, the list is not reversed.